domains/transfer-agreements/TransferAgreements.API/API.IntegrationTests/Data/TransferAgreementRepositoryTest.cs

```diff
@@ -57,33 +50,40 @@ private static List<TransferAgreement> SetupTransferAgreements(Guid senderId)
         {
             new()
             {
+                Id = Guid.NewGuid(),
                 StartDate = DateTimeOffset.UtcNow,
                 EndDate = DateTimeOffset.UtcNow.AddDays(10),
                 SenderId = senderId,
                 SenderName = "nrgi A/S",
                 SenderTin = "44332211",
                 ReceiverTin = "12345678",
                 ReceiverReference = Guid.NewGuid(),
+                TransferAgreementNumber = 1
             },
             new()
             {
+                Id = Guid.NewGuid(),
                 StartDate = DateTimeOffset.UtcNow,
                 EndDate = DateTimeOffset.UtcNow.AddDays(10),
                 SenderId = senderId,
                 SenderName = "nrgi A/S",
                 SenderTin = "44332211",
                 ReceiverTin = "12345678",
                 ReceiverReference = Guid.NewGuid(),
+                TransferAgreementNumber = 2
             },
             new()
             {
+                Id = Guid.NewGuid(),
                 StartDate = DateTimeOffset.UtcNow,
                 EndDate = DateTimeOffset.UtcNow.AddDays(10),
                 SenderId = senderId,
                 SenderName = "nrgi A/S",
                 SenderTin = "44332211",
                 ReceiverTin = "12345678",
                 ReceiverReference = Guid.NewGuid(),
+                TransferAgreementNumber = 3
+
```

Comment:
The way we are testing here, is without the validations included, which we are skipping because we are seeding the database directly by using SQL. All the validations exist on the service layer. I understand that the goal is to test whether we are triggering the Unique key Contraint, but wouldn't it be better to include validations as well? in a real use case, if a user attempted to create multiple Transfer Agreements, with the same dates, he would be rejected before the request even reaches the repository, let alone the database. Then the Duplicate key violation would never trigger. So this integration test serves to make sure the database is concurrency safe, but data integrity, has not been accounted for, since we are omitting all validations.